docs/src/pages/styles/advanced/advanced.md

```diff
@@ -553,6 +553,32 @@ This vulnerability would allow the attacker to execute anything. However, with a
 
 You can read more about CSP on the [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP).
 
+### Does Material UI support CSP?
+
+Most Material UI components support a strict CSP. Specifically, this means you will not need to use `'unsafe-inline'`. However, some components still use inline styles. If you use these components you must set `'unsafe-inline'` in the `style-src` directive.
+
+As of this writing, the components known to use inline styles include:
```

Comment:
I don't think that we should go into details here, better keep it simple on the docs. It will get outdated.